types/models.d.ts

```diff
@@ -184,42 +182,30 @@ declare module 'mongoose' {
     collection: Collection;
 
     /** Creates a `count` query: counts the number of documents that match `filter`. */
-    count(callback?: Callback<number>): QueryWithHelpers<number, THydratedDocumentType, TQueryHelpers, TRawDocType>;
-    count(filter: FilterQuery<TRawDocType>, callback?: Callback<number>): QueryWithHelpers<number, THydratedDocumentType, TQueryHelpers, TRawDocType>;
+    count(): QueryWithHelpers<number, THydratedDocumentType, TQueryHelpers, TRawDocType>;
```

Comment:
can this function be called without any parameters? the JSDoc mentions that `filter` [is required](https://mongoosejs.com/docs/api/model.html#model_Model-count)
*though this may just be outdated information?

btw, maybe also consider adding a deprecation notice to the TSDoc for this function

Reply:
It is perfectly fine to call `count()` without any parameters. I'll make `filter` optional in the jsdoc.